projects/plugins/jetpack/modules/search/customberg/hooks/use-search-options.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { useMemo } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+
+/**
+ * Fetches values and setters for various search configuration values.
+ *
+ * @returns {object} values and setters
+ */
+export default function useSearchOptions() {
+	const [ theme, setTheme ] = useEntityProp( 'root', 'site', 'jetpack_search_color_theme' );
+	const [ resultFormat, setResultFormat ] = useEntityProp(
+		'root',
+		'site',
+		'jetpack_search_result_format'
+	);
+	const [ sort, setSort ] = useEntityProp( 'root', 'site', 'jetpack_search_default_sort' );
+	const [ trigger, setTrigger ] = useEntityProp( 'root', 'site', 'jetpack_search_overlay_trigger' );
+	const [ color, setColor ] = useEntityProp( 'root', 'site', 'jetpack_search_highlight_color' );
+	const [ sortEnabled, setSortEnabled ] = useEntityProp(
+		'root',
+		'site',
+		'jetpack_search_enable_sort'
+	);
+	const [ infiniteScroll, setInfiniteScroll ] = useEntityProp(
+		'root',
+		'site',
+		'jetpack_search_inf_scroll'
+	);
+	const [ showLogo, setShowLogo ] = useEntityProp(
+		'root',
+		'site',
+		'jetpack_search_show_powered_by'
+	);
+
+	const [ excludedPostTypesCsv, setExcludedPostTypesCsv ] = useEntityProp(
+		'root',
+		'site',
+		'jetpack_search_excluded_post_types'
+	);
+	// Excluded Post Types is stored as a CSV string in site options. Convert into
+	const excludedPostTypes = useMemo( () => excludedPostTypesCsv?.split( ',' ), [
```

Comment:
As the initial value for `jetpack_search_excluded_post_types` could be an empty string, when splitting on which, it produces `[""]` and breaks the size limit check.
![image](https://user-images.githubusercontent.com/1425433/127250687-8e7682d1-242a-42df-8d4d-d843fff3bffb.png)

Reply:
Good catch, fixed.